### DISCUSSION TITLE
Question about setting the SignalTime of ElapsedEventArgs

### DISCUSSION THREAD

**Discussion by LeaFrock** [dotnet/runtime#44785] [Q&A] [answered]
When I read the source codes of **System.Timers.Timer**,  I notice this [line](
https://github.com/dotnet/runtime/blob/master/src/libraries/System.ComponentModel.TypeConverter/src/System/Timers/Timer.cs#L293):

`ElapsedEventArgs elapsedEventArgs = new ElapsedEventArgs(DateTime.UtcNow.ToFileTime());`

And in the [constructor ](https://github.com/dotnet/runtime/blob/master/src/libraries/System.ComponentModel.TypeConverter/src/System/Timers/ElapsedEventArgs.cs) of ElapsedEventArgs:

`SignalTime = DateTime.FromFileTime(fileTime);`

Is that necessary or should be optimized? Any comments welcomed.

**Comment by stephentoub** [comment, marked as answer]
This is a holdover from how it was constructed in .NET Framework.  It could be changed to just pass DateTime.UtcNow and have the constructor accept and store a DateTime, rather than roundtripping through filetime.

**Reply by LeaFrock** [reply]
Well, need I make a pull request? Or you will handle that?

**Reply by stephentoub** [reply]
Please feel free to do so.  Thanks.
